spec/report_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'spec_helper'
+
+describe EasyPost::Report do
+  [:shipment, :tracker, :payment_log].each do |report_type|
+    context "report" do
+      let(:type) { report_type }
+      let(:type_factory) {
+        case type
+        when :shipment
+          "ShipmentReport"
+        when :tracker
+          "TrackerReport"
+        when :payment_log
+          "PaymentLogReport"
+        else
+          raise "Invalid type!"
+        end
+      }
+
+      describe '#create' do
+        it 'creates a report object' do
+          report = EasyPost::Report.create(
+                                           start_date: Date.today - 30,
+                                           end_date: Date.today,
+                                           type: type
+                                           )
+          expect(report.object).to eq type_factory
+          # will switch that test on when tracker status is deployed
+          # expect(report.status).to eq 'available'
+        end
+      end
+
+      describe '#retrieve' do
+        it 'retrieves a user created report by public_id' do
+          report_1 = EasyPost::Report.create(
+                                             start_date: Date.today - 30,
+                                             end_date: Date.today,
+                                             type: type
+                                             )
+          report_2 = EasyPost::Report.retrieve(public_id: report_1.id)
+
+          expect(report_1.object).to eq type_factory
+          expect(report_2.object).to eq type_factory
+          expect(report_2.id).to eq(report_1.id)
+        end
+      end
+
+      describe '#all' do
+        it 'retrieves all user created reports' do
```

Comment:
Okay I think I've determined why these tests failed: it would seem that they only work the first time they are run each day, which explains why they _seemed_ to pass for you locally. Specifically, the Shipment and PaymentLog reports fail here because they do not pass the criteria to be regenerated; the Tracker report auto-regenerates so it isn't affected. The other tests set the end_date to today, which triggers the report to regenerate (which doesn't happen here since the report's end_date is from before today). As far as solutions, it looks like you could either make the end_dates today so its like the other tests, or maybe use DateTime instead of Date in these tests so that you get more precise timestamps and therefore trigger a new report each time. Either of those solutions should be viable, though I haven't tested them.